server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateServiceTests.java

```diff
@@ -882,7 +896,7 @@ public void testGetValidPreviousClusterUUIDWithMultipleChains() throws IOExcepti
             "cluster-uuid3",
             "cluster-uuid1"
         );
-        mockObjectsForGettingPreviousClusterUUID(clusterUUIDsPointers);
+        mockObjectsForGettingPreviousClusterUUID(clusterUUIDsPointers, randomBoolean());
```

Comment:
if this is random, how would you ensure previous matches uuid-3 for the assertion below.